apps/storybook/components/funke/IdCardPinScreen.stories.tsx

```diff
@@ -0,0 +1,243 @@
+    const firstFocusableIndex = pin.findIndex((digit) => digit === '')
+    return firstFocusableIndex === -1 || index <= firstFocusableIndex
+  }
+
+  const setPinValue = (digit: string, index: number) => {
+    const sanitized = digit.replace(/[^0-9]/g, '')
+    if (sanitized === '') return
+
+    setPin((pin) => {
+      const newPin = [...pin]
+      newPin[index] = sanitized
+      return newPin
+    })
+
+    // Focus next input
+    if (index < pin.length - 1) {
+      programmaticFocus.current = true
+      inputRefs.current[index + 1]?.focus()
+    } else if (index === pin.length - 1) {
+      inputRefs.current[index]?.blur()
+      onPinComplete(pin.join(''))
+    }
+  }
+
+  const focusInput = (index: number) => {
+    // We programmatically focus, but the user can also focus. This is a hack to
+    // know when we are in programmatic mode
+    if (programmaticFocus.current) {
+      programmaticFocus.current = false
+      return
+    }
+
+    // If e.g. the user presses the 3rd pin, but 2nd pin is still
+    // empty it will focus the 2nd pin
+    const firstEmptyIndex = pin.findIndex((digit) => digit === '')
+    if (firstEmptyIndex !== -1 && firstEmptyIndex < index) {
+      inputRefs.current[firstEmptyIndex]?.focus()
+    } else {
+      inputRefs.current[index]?.focus()
+    }
+  }
+
+  return (
+    <Page gap="$6" justifyContent="space-between">
+      <ProgressBar value={25} />
+      <YStack flexBasis={0} flexGrow={2} flexShrink={1} justifyContent="space-between">
+        <Heading variant="title">{title}</Heading>
+        <XStack display={state === 'enterPin' ? 'flex' : 'none'} gap="$3" justifyContent="center">
+          {pin.map((digit, index) => (
+            // biome-ignore lint/suspicious/noArrayIndexKey: index is the correct key here
+            <YStack key={index} maxWidth={35} flex-1 justifyContent="center">
+              <Input
+                p="$0"
+                textAlign="center"
+                borderWidth={0}
+                fontSize="$6"
+                ref={(ref) => {
+                  inputRefs.current[index] = ref
+                }}
+                value={digit}
+                focusable={isIndexFocusable(index)}
+                onFocus={() => focusInput(index)}
+                onKeyPress={(text) => setPinValue(text.nativeEvent.key, index)}
+                focusVisibleStyle={{
+                  outlineStyle: 'none',
+                }}
+                outlineStyle="none"
+                autoFocus={index === 0}
+                maxLength={1}
+                inputMode="numeric"
+                accessible={true}
+                aria-label={`Pin digit ${index + 1}`}
+                accessibilityHint="Enter the digit of the pin"
+              />
+              <Stack borderBottomWidth={1} borderBottomColor="$grey-900" width="100%" />
+            </YStack>
+          ))}
+        </XStack>
+      </YStack>
+      <Stack gap="$4" flexBasis={0} flexGrow={4} flexShrink={1}>
```

Comment:
For the layout you use a lot of different flex methods which can become quite a mess if you want to structure a page layout.

My approach is always using justify-content to place the main elements of a stack (which you do nicely). And then use a combination of padding and gap to space the content elements. 

I would combine the progress bar and the title into 1 re-usable header for all the pages. And then have one YStack as the parent for the rest of your content. You can give this a flex grow of 1 to make it scale to all the space that is left on your screen.
Ideally you set the boundaries of your content on this stack, so probably a padding of '4' to make it stand out from the sides.

Then you have a stack with the pin input and the pin card. The spacing for these elements can be done with gap to make theme separated from each other. 

So then you only need flex grow and gap and sometimes some padding.

Reply:
It might also be preference but I think it's much easier to read and debug then 3 different flex methods on every stack that is used. 

I think as long as you have the base layout (your main elements spread out using 'justify-content' and a base content stack with 'flex-grow') then you can still use it for your inner content.

Reply:
I spent an enormous amount of time on this, and there's a reason I did it exactly this way, as otherwise the content would jump around. 

I even opened an issue in tamagui for it: https://github.com/tamagui/tamagui/issues/2806

Reply:
It seems it was an actual bug, and they already fixen my issue :)

Reply:
Okay cleaned it up quite a bit now. Based on your approach